distributions/test/src/test/java/com/connexta/ion/replication/distributions/test/IntegrationTest.java

```diff
@@ -0,0 +1,149 @@
+          .withNetworkAliases("replication-solr")
+          .withExposedPorts(SOLR_PORT);
+
+  @ClassRule
+  public static GenericContainer replication =
+      new GenericContainer("r.ion.phx.connexta.com/replication:0.3.0-SNAPSHOT")
+          .withNetwork(network)
+          .withNetworkAliases("ion-replication")
+          .withClasspathResourceMapping(
+              "serverKeystore.jks",
+              "/opt/replication/config/keystores/serverKeystore.jks",
+              BindMode.READ_ONLY)
+          .withClasspathResourceMapping(
+              "serverTruststore.jks",
+              "/opt/replication/config/keystores/serverTruststore.jks",
+              BindMode.READ_ONLY)
+          .withClasspathResourceMapping(
+              "ssl.properties", "/opt/replication/config/ssl.properties", BindMode.READ_ONLY)
+          .withClasspathResourceMapping(
+              "application.yml", "/opt/replication/config/application.yml", BindMode.READ_ONLY);
+
+  public static String solrUrl;
+
+  public static String sourceUrl;
+
+  public static String destinationUrl;
+
+  public static InputStream getFileAsStream(String filePath) {
+    return IntegrationTest.class.getClassLoader().getResourceAsStream(filePath);
+  }
+
+  /**
+   * Retrieves a file and returns the contents within as a string. The filePath is assumed to be
+   * relative to the {@link IntegrationTest} class.
+   *
+   * @param filePath the path of the file relative to the {@link IntegrationTest} class
+   * @return A String containing the contents of the file
+   */
+  public static String getFileAsText(String filePath) {
+    return getFileAsText(filePath, ImmutableMap.of());
+  }
+
+  /**
+   * Retrieves a file and returns the contents within as a string. The filePath is considered
+   * relative to the class provided. Text in the file can be marked for variable replacement by
+   * giving it the format: $variableName$ . The params should then contain a mapping of variableName
+   * to the desired replacement value.
+   *
+   * @param filePath The path of the file relative to the class provided
+   * @param params A map of variables in the resource file and their replacement values
+   * @return A String containing the contents of the file with any variables replaced
+   */
+  @SuppressWarnings({
+    "squid:S00112" /* A generic RuntimeException is perfectly reasonable in this case. */
+  })
+  public static String getFileAsText(String filePath, ImmutableMap<String, String> params) {
+
+    StringSubstitutor strSubstitutor = new StringSubstitutor(params);
+
+    strSubstitutor.setVariablePrefix(RESOURCE_VARIABLE_DELIMETER);
+    strSubstitutor.setVariableSuffix(RESOURCE_VARIABLE_DELIMETER);
+    String fileContent;
+
+    try {
+      fileContent =
+          IOUtils.toString(
+              IntegrationTest.class.getClassLoader().getResourceAsStream(filePath), "UTF-8");
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to read filepath: " + filePath);
```

Comment:
AssertionError seems a little misleading here since we're not actually making an assertion. The test will still fail so I would rather keep this as is.

Reply:
Sure. Then let's go with either `UncaughtIOException` or `IOError` or `IllegalStateException` since this is something that shouldn't have happen. `RuntimeException` shouldn't be thrown. If static analysis is warning you, and you are forced to put a comment about it, then it means something else should be thrown. Unless you are forced by existing libraries, that declared they threw that (when they should have never done that), just don't.